packages/astro/src/runtime/client/hmr.ts

```diff
@@ -1,46 +1,7 @@
 /// <reference types="vite/client" />
 
 if (import.meta.hot) {
-	// Vite injects `<style data-vite-dev-id>` for ESM imports of styles
-	// but Astro also SSRs with `<style data-astro-dev-id>` blocks. This MutationObserver
-	// removes any duplicates as soon as they are hydrated client-side.
-	const injectedStyles = getInjectedStyles();
-	const mo = new MutationObserver((records) => {
-		for (const record of records) {
-			for (const node of record.addedNodes) {
-				if (isViteInjectedStyle(node)) {
-					injectedStyles.get(node.getAttribute('data-vite-dev-id')!)?.remove();
-				}
-			}
-		}
-	});
-	mo.observe(document.documentElement, { subtree: true, childList: true });
-
-	// Vue `link` styles need to be manually refreshed in Firefox
-	import.meta.hot.on('vite:beforeUpdate', async (payload) => {
```

Comment:
I manually tested this and it seems to not apply anymore, so I removed it. The rest of this file relates to our CSS handling, which is no longer needed as Vite handles all now, so pretty much the entire file is empty and not need to be loaded anymore.